tests/cacl/test_cacl_application.py

```diff
@@ -835,6 +835,83 @@ def generate_scale_rules(duthost, ip_type):
     duthost.command("iptables -I INPUT 3 -p tcp -m tcp --dport 22 -j ACCEPT")
 
 
+def verify_cacl_show_acl_rule(duthost):
+    """
+    Converts the CACL rules in the generated SCALE_ACL_FILE into a dict,
+    and verifies that they match the applied acl rules through the `show acl
+    rule` command.
+    """
+
+    acl_json = duthost.command(f"cat {SCALE_ACL_FILE}")["stdout_lines"]
+    acl_rules_expected_dict = json.loads("".join(acl_json))
+
+    acl_sets_extracted = acl_rules_expected_dict["acl"].get("acl-sets", {}).get("acl-set", {})
+    acl_rules_expected = {}
+
+    # Reconstruct the dict such that we can index easily using the actual rules
+    for acl_table, acl_table_conf in acl_sets_extracted.items():
+        # Reformat name as produced by acl-loader
+        acl_table_name = acl_table.upper().replace("-", "_")
+        acl_table_rules = {}
+
+        for acl_entry_num, acl_entry_conf in acl_table_conf['acl-entries']['acl-entry'].items():
+            acl_entry_name = f"RULE_{acl_entry_num}"
+            acl_table_rules[acl_entry_name] = acl_entry_conf
+
+        acl_rules_expected[acl_table_name] = acl_table_rules
+
+    logger.debug(f"Expected rules: {acl_rules_expected}")
+
+    acl_rules_actual = get_cacl_tables_and_rules(duthost)
+
+    logger.debug(f"Actual rules: {acl_rules_actual}")
+
+    incorrect_rules = []
+    incorrect_table_rule_count = []
+
+    # Check each rule in each ACL table on the DUT
+    for acl_table in acl_rules_actual:
+        acl_table_name = acl_table["name"].strip()
+        acl_table_rules = acl_table["rules"]
+        acl_table_expected_rules = acl_rules_expected[acl_table_name]
+
+        # Ensure that the expected number of rules are present in each table
+        if len(acl_table_rules) != len(acl_table_expected_rules):
+            logger.warning(f"{acl_table_name}: Incorrect amount of rules for table")
+            logger.warning(f"{acl_table_name}: {len(acl_table_rules)} != {len(acl_table_expected_rules)}")
+            incorrect_table_rule_count.append(acl_table_name)
+
+        # Ensure that each rule is configured correctly
+        for acl_rule in acl_table["rules"]:
+            valid_rule = True
+            acl_rule_name = acl_rule["name"].strip()
+            rule_expected_conf = acl_table_expected_rules[acl_rule_name]
+
+            if acl_rule["SRC_IP"] != rule_expected_conf["ip"]["config"]["source-ip-address"]:
+                valid_rule = False
+
+            if acl_rule["action"] != rule_expected_conf["actions"]["config"]["forwarding-action"]:
+                valid_rule = False
+
+            if int(acl_rule["priority"]) != (10000 - int(rule_expected_conf["config"]["sequence-id"])):
+                valid_rule = False
+
+            if not valid_rule:
+                logger.warning(f"{acl_table_name}: Config '{acl_rule}' did not match expected '{rule_expected_conf}'")
+                incorrect_rules.append(acl_rule)
```

Comment:
If you have a list to show what issues are detected, how about to add the issue description into the issue list, not only the acl rule name? It can give us more detailed information.

Reply:
Good idea, I've enhanced the errors so that the specific issues with rules print:
```
E       Incorrectly configured ACL rules found:
E       NTP_ACL: The following rules had issues:
E         - RULE_2: Expected IP 20.0.0.3/32 does not match WRONG IP
```